runtime/srutils.c

```diff
@@ -745,6 +754,10 @@ void seedRandomNumber(void)
 	seedRandomInsecureNumber();
 }
 
+void closeRandomNumberFd(void)
```

Comment:
This looks like it (and all calls) can simply be removed?

Reply:
Yes, it's not strictly needed since the file descriptor would already be included in the list closed by `prepareBackground()`. However, I was concerned that `seedRandomNumber()` could be changed in future to only open `/dev/urandom` if `fdURandom` was negative. Which would then re-introduce the bug.

I could remove `closeRandomNumberFd()` and instead add a comment to `seedRandomNumber()` warning of this, if you'd prefer?

Reply:
> if `fdURandom` was negative

as we know about this problem, I think it would make sense to set it to -1 on close. That way we could actually do the -1 check.

side-note: think about refactoring. If we would leave the code as-is, it would sooner or later be removed because it "obviously" is unnecessary.

Reply:
OK, so perhaps the following would be a neater fix:

* Have `seedRandomNumber()` only open `/dev/urandom` if  `fdURandom` is negative.
* Add a new `seedRandomNumberForChild()` function, which simply sets `fdURandom` to -1 and calls `seedRandomNumber()` again.
* Have `prepareBackground()` call `seedRandomNumberForChild()` after closing file descriptors.

That would address my concerns, and means we wouldn't have `closeRandomNumberFd()` (which as you say could be removed by future refactoring work). If this sounds reasonable to you, then I'll update the PR to implement this today.

Reply:
sounds good to me!

Reply:
Excellent, thanks. I've now updated this PR.